app/src/main/kotlin/github/com/rhacco/dota2androidapp/activities/LiveMatchesActivity.kt

```diff
@@ -29,21 +39,26 @@ class LiveMatchesActivity : BaseLifecycleActivity<MatchesViewModel>() {
     }
 
     // Add tournament matches first, then sort by average MMR (descending)
-    private fun updateLiveMatchesListData(listItemData: LiveMatchesListAdapter.ListItemData) {
-        if (listItemData.mAverageMMR < 1) {
-            mListAdapter.mListItemsData.add(0, listItemData)
+    private fun updateLiveMatchesListData(newItem: LiveMatchesListAdapter.ListItemData) {
+        // Strangely, sometimes the same match arrives here twice, so prevent creating duplicates
+        mListAdapter.mListItemsData
```

Comment:
it looks like you're only adding on matches that haven't previously been shown and invalidating the list after you add on the new matches.

UI wise, this is very expensive. You could do the same thing with recyclerview and do a notifyItemAdded() to save on the UI rendering.

Next time you test this, look for the "Frames dropped" message on Logcat. Every 16ms, Android refreshes the screen and if there is too much work on the UI thread, it will skip that refresh. This can lead to a laggy user experience. For our intents and purposes, we shouldn't see that since this app is very simple at this point.

Also, why don't we just invalidate all the matches and just show the most recent ones? What benefit do we get  from adding only the matches not shown in terms of user experience? I kind of understand where you're going but I just want to know why.

Reply:
Currently I'm doing a full wipe and new fetch of all live matches. With the migration to RecyclerView in #9 I will use notifyItemInserted() later, instead of wiping everything